### PR TITLE
docs: support new Apple Silicon homebrew path in setup instructions

### LINKED ISSUE #4396: [Docs]: development env setup instructions don't support Apple Silicon
### Is there an existing issue already for your request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new documentation enhancement to be evaluated.

### What problem in the existing documentation this issue aims to solve?

The setup instructions in `contribute/development_environment/README.md` contain a fragment setting up the paths for binaries, man and libraries of various packages installed by homebrew.

These paths don't work for Apple Silicon, where homebrew has adopted a new default path.

### Describe what additions need to be done to the documentation

We need to indicate the difference between Intel and Apple Silicon to ensure the paths are correct.

### Describe what pages need to change in the documentation, if any

`contribute/development_environment/README.md`

### Describe what pages need to be removed from the documentation, if any

_No response_

### Additional context

Discussion in Homebrew community
https://github.com/Homebrew/brew/issues/9177

### Backport?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jsilvela**
Closes #4396

### CODE REVIEW

**Review by litaocdl — APPROVED** [review]

File: contribute/development_environment/README.md [hunk 1]
```diff
@@ -116,7 +116,14 @@ brew install jq \
 ```
 
 You can then follow the provided instructions. As you'll see, you need to add
-the following lines to the profile of your shell (eg `~/.bash_profile`):
+the following lines to the profile of your shell (eg `~/.bash_profile` or
+`~/.zlogin`):
```
Comment (GabriFedi97) [inline contribute/development_environment/README.md:120]:
```suggestion
`~/.zprofile`):
```

**Review by GabriFedi97 — APPROVED** [review]
LGTM

File: contribute/development_environment/README.md [hunk 2]
```diff
@@ -116,30 +116,40 @@ brew install jq \
...
-the following lines to the profile of your shell (eg `~/.bash_profile`):
+the following lines to the profile of your shell (eg `~/.bash_profile` or
+`~/.zprofile`):
+
+**Warning**: for Apple Silicon macs, the new default location for homebrew
+is `/opt/homebrew`, not `/usr/local`. See [discussion](https://github.com/Homebrew/brew/issues/9177).
+Please choose the right definition of `HOMEBREW_PATH` for the following section:
+
+* Intel:  `export HOMEBREW_PATH="/usr/local"`
+* Apple Silicon: `export HOMEBREW_PATH="/opt/homebrew"`
```
Comment (mnencia) [inline contribute/development_environment/README.md:127]:
```
BREW_PREFIX=$(brew --prefix)
```
does the detection.

It also doesn't need to be exported, and probably can be used at the end of the code snipplet.

Comment (jsilvela) [reply contribute/development_environment/README.md]: ah, saw this after my own correction.. Thanks

**Review by mnencia — APPROVED** [review]

**Pull request merged**